### PR TITLE
Partial Jsonization of Mood Indicators

### PR DESCRIPTION
**Pull request by Saicchi**
#### Summary
Infrastructure "Partial Jsonization of Mood Indicators"

#### Purpose of change
Allow more mood indicators of other thresholds.

#### Describe the solution
Add a loader for `mood_face`, but `mood_face` must have an specific `id` or else it won't be used.
`DEFAULT` and `DEFAULT_HORIZONTAL` for the default value, must not be deleted ( modifying is fine ).

`THRESH_TRAIT` and `THRESH_TRAIT_HORIZONTAL` for traits.
Examples:
For `THRESH_BIRD`: `THRESH_BIRD` and `THRESH_BIRD_HORIZONTAL`
For `THRESH_SPIDER`: `THRESH_SPIDER` and `THRESH_SPIDER_HORIZONTAL`

The `id` must be exact as it is hardcoded to look for that.

```cpp
{
    "type": "mood_face",
    "id": "DEFAULT_HORIZONTAL",
    "values": [ // mandatory
        {
            "value": 200, // mandatory
            "face": "<color_green>@w@</color>" // mandatory
        },
        {
            "value": -200,
            "face": "<color_red>XvX</color>" // adding a color is also mandatory
        },
        {
            "value": -201, // morale is clamped [200, -200] on regular gameplay, not necessary unless debugging
            "face": "<color_yellow>@^@</color>"
        }
    ]
}
```
#### Describe alternatives you've considered
None

#### Testing
- [x] Tested `DEFAULT` and `DEFAULT_HORIZONTAL`
- [x] Tested `BIRD`, `URSINE` and `FELINE` thresholds

#### Additional context
During regular gameplay morale is clamped at [ 200, -200 ], but with debugging tools it can go well past that.

### CODE REVIEW

File: src/mood_face.cpp [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+    if( mood_face_horizontal ) {
+        face_type.append( "_HORIZONTAL" );
+    }
+
+    // Valid: THRESH_MUT and THRESH_MUT_HORIZONTAL
+    // Example: THRESH_BIRD and THRESH_BIRD_HORIZONTAL
+    for( const mood_face &face_id : mood_face::get_all() ) {
+        if( face_id.getId().str() == face_type ) {
+            mood_face_cache = face_id.getId();
```
Comment (BrettDong) [inline src/mood_face.cpp:94]:
```suggestion
            mood_face_cache = face_id.getId();
            break;
```

**Comment by BrettDong** [comment]
Regarding the Text Changes Analyzer failure, you can add `mood_face` to the list of `ignorable` in `lang/extract_json_strings.py` at line 80.

**Comment by wapcaplet** [comment]
@Saicchi I really don't like that you had to delete the `const` qualifiers from several `widget` methods. These are intended to be display-only functions that guarantee they do not modify the `avatar` instance. This change has complicated something I am trying to do on #51438 as well.

Would it be possible to make a version of these mood indicator functions that don't modify the `avatar` instance? Ultimately I would like to migrate these mood indicators to widgets as well (it has been part of my plan from the beginning but I never spent much energy trying to tackle it).

**Comment by Saicchi** [comment]
It's been a while since I last looked at this so I don't remember what I wrote all that well. Still I think the reason I removed `const` was to cache the mood face in the `avatar` class, the cache could me moved out of it into a static variable I believe.

**Pull request merged**